src/lib/nvme_pt_ioctl.c

```diff
@@ -307,27 +307,31 @@ static int opal_level0_disc_pt(struct sed_device *device)
 			curr_feat = &disc_data->feats[feat_no];
 			curr_feat->type = feat_code;
 			cpy_tper_feat(discv, &desc->feat.tper.flags);
+			discv->feat_avail_flag.feat_tper = 1;
 
 			feat_no++;
 			break;
 		case OPAL_FEAT_LOCKING:
 			curr_feat = &disc_data->feats[feat_no];
 			curr_feat->type = feat_code;
 			cpy_locking_feat(discv, &desc->feat.locking.flags);
+			discv->feat_avail_flag.feat_locking = 1;
 
 			feat_no++;
 			break;
 		case OPAL_FEAT_GEOMETRY:
 			curr_feat = &disc_data->feats[feat_no];
 			curr_feat->type = feat_code;
 			cpy_geometry_feat(discv, &desc->feat.geo);
+			discv->feat_avail_flag.feat_geometry = 1;
 
 			feat_no++;
 			break;
 		case OPAL_FEAT_DATASTORE:
 			curr_feat = &disc_data->feats[feat_no];
 			curr_feat->type = feat_code;
 			cpy_datastr_feat(discv, &desc->feat.datastr.datastr_tbl);
+			discv->feat_avail_flag.feat_datastr_table = 1;
```

Comment:
why don't we want also to have flag for OPAL_FEAT_SUM?

Reply:
Currently we are not printing the SUM feature.
That goes into a separate patch. Will include flag in it.

Reply:
Flag is intended to indicate if specific discovery header is present. Please add it into this patch series (no need to add display of SUM feature)

Reply:
will do it